GeoLib/AnalyticalGeometry.h

```diff
@@ -60,11 +60,28 @@ Orientation getOrientation (const GeoLib::Point* p0,
  * @param plane_normal the normal of the plane the polygon is located in
  * @param d parameter from the plane equation
  */
-void getNewellPlane (const std::vector<GeoLib::Point*>& pnts,
+template <class T_POINT>
+void getNewellPlane (const std::vector<T_POINT*>& pnts,
```

Comment:
You are right, we should rename the method to something like `getSupportingPlane`. The algorithm uses Newell's method, described in the book "Real-Time Collision Detection" from Christer Ericson. This fact is also missing in the documentation.
@norihiro-w Should I update the branch or should we make a separat PR?